airflow/exceptions.py

```diff
@@ -107,6 +107,10 @@ class AirflowOptionalProviderFeatureException(AirflowException):
     """Raise by providers when imports are missing for optional provider features."""
 
 
+class AirflowInternalRuntimeError(BaseException):
```

Comment:
Should this subclass from `RuntimeError` instead?

Reply:
`RuntimeError` and subclasses as well as `Exception` catch somewhere during Non-DB preparation, in some cases it is do not affect tests cases in the other make them flakey. This one decoupled into the separate PR: https://github.com/apache/airflow/pull/38778 - the main idea, if test somewhere access to the DB when it prohibited it should failed instead of ignore it and go next, or marked as DB test

There is also different things which I've found during this attempt, some of them already merged some of them wait their time